### ISSUE TITLE
Fails to create new daily note

### ISSUE THREAD

**Issue by fkjogu** [liamcain/obsidian-calendar-plugin#34]
When clicking on a day without an associated note, and clicking `create`, the plugin fails to create a new page with the following error. Using Obsidian Version v0.9.11 on Windows 10 and plugin version v1.2.2.

```
app.js:1 Uncaught (in promise) TypeError: Cannot read property 'cachedData' of null
    at t.<anonymous> (app.js:1)
    at app.js:1
    at Object.next (app.js:1)
    at app.js:1
    at new Promise (<anonymous>)
    at s (app.js:1)
    at t.cachedRead (app.js:1)
    at getDailyNoteTemplateContents (eval at <anonymous> (app.js:1), <anonymous>:1091:22)
    at createDailyNote (eval at <anonymous> (app.js:1), <anonymous>:1112:36)
    at CalendarView._createDailyNote (eval at <anonymous> (app.js:1), <anonymous>:1239:33)
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
s @ app.js:1
t.cachedRead @ app.js:1
getDailyNoteTemplateContents @ template.ts:41
createDailyNote @ template.ts:68
_createDailyNote @ view.ts:134
onAccept @ view.ts:148
eval @ modal.ts:30
async function (async)
eval @ modal.ts:30

```

**Comment by liamcain** [comment]
@fkjogu Hi, thanks for the report! I think I have a fix, but just to be sure, could you let me know what you have configured as your "Template File Location" in settings? 

<img width="921" alt="image" src="https://user-images.githubusercontent.com/693981/98815610-672e9d80-23f5-11eb-9518-a06eb69e0aff.png">

**Comment by fkjogu** [comment]
Thanks for looking into this. Here is my configuration for both `Templates` and `Daily notes`

![image](https://user-images.githubusercontent.com/13063485/98920572-8b12e180-24d0-11eb-9d22-2a38e499bed6.png)

![image](https://user-images.githubusercontent.com/13063485/98920690-b0075480-24d0-11eb-9dbf-a3c306500c76.png)

**Comment by iris2jane** [comment]
Plugin version v1.2.3 on macOS cannot create new daily note either.

**Comment by berdan** [comment]
Different error than the one above though. (1.2.3 on Obsidian 0.9.13, Windows)

```
settings.ts:32 Uncaught (in promise) TypeError: Cannot read property 'trim' of undefined
    at getDailyNoteTemplatePath (eval at <anonymous> (app.js:1), <anonymous>:347:53)
    at getDailyNoteTemplateContents (eval at <anonymous> (app.js:1), <anonymous>:1082:34)
    at createDailyNote (eval at <anonymous> (app.js:1), <anonymous>:1112:36)
    at CalendarView._createDailyNote (eval at <anonymous> (app.js:1), <anonymous>:1239:33)
    at CalendarView._openOrCreateDailyNote (eval at <anonymous> (app.js:1), <anonymous>:1222:28)
    at Array.click_handler (eval at <anonymous> (app.js:1), <anonymous>:985:3)
    at HTMLTableCellElement.click_handler (eval at <anonymous> (app.js:1), <anonymous>:545:35)
getDailyNoteTemplatePath @ settings.ts:32
getDailyNoteTemplateContents @ template.ts:27
createDailyNote @ template.ts:64
_createDailyNote @ view.ts:134
_openOrCreateDailyNote @ view.ts:114
click_handler @ Calendar.svelte:232
click_handler @ Calendar.svelte:237
async function (async)
_openOrCreateDailyNote @ view.ts:114
click_handler @ Calendar.svelte:232
click_handler @ Calendar.svelte:237
```

**Comment by liamcain** [comment]
Hi, thanks for the reports! I just released v1.2.4 which now properly find the daily note template and fix this issue. Please let me know if the issue persists~

**Comment by berdan** [comment]
Works fine for me now.

**Comment by fkjogu** [comment]
Creating the note now works for me, but it has issues filling in the template. I get

```
template.ts:43 Failed to read daily note template 'templates/daily.md' TypeError: Cannot read property 'cachedData' of null
    at t.<anonymous> (app.js:1)
    at app.js:1
    at Object.next (app.js:1)
    at app.js:1
    at new Promise (<anonymous>)
    at s (app.js:1)
    at t.cachedRead (app.js:1)
    at getDailyNoteTemplateContents (eval at <anonymous> (app.js:1), <anonymous>:1091:38)
    at createDailyNote (eval at <anonymous> (app.js:1), <anonymous>:1113:36)
    at CalendarView._createDailyNote (eval at <anonymous> (app.js:1), <anonymous>:1240:33)
getDailyNoteTemplateContents @ template.ts:43
async function (async)
getDailyNoteTemplateContents @ template.ts:40
createDailyNote @ template.ts:67
_createDailyNote @ view.ts:134
onAccept @ view.ts:148
eval @ modal.ts:30
template.ts:43 Failed to read daily note template 'templates/daily.md' TypeError: Cannot read property 'cachedData' of null
    at t.<anonymous> (app.js:1)
    at app.js:1
    at Object.next (app.js:1)
    at app.js:1
    at new Promise (<anonymous>)
    at s (app.js:1)
    at t.cachedRead (app.js:1)
    at getDailyNoteTemplateContents (eval at <anonymous> (app.js:1), <anonymous>:1091:38)
    at createDailyNote (eval at <anonymous> (app.js:1), <anonymous>:1113:36)
    at CalendarView._createDailyNote (eval at <anonymous> (app.js:1), <anonymous>:1240:33)
getDailyNoteTemplateContents @ template.ts:43
async function (async)
getDailyNoteTemplateContents @ template.ts:40
createDailyNote @ template.ts:67
_createDailyNote @ view.ts:134
onAccept @ view.ts:148
eval @ modal.ts:30

```

**Comment by liamcain** [comment]
@fkjogu Thanks for the follow-up. Looks like Obsidian isn't able to find the template in your vault. Can you just double-check that the path in settings is right and the file exists? (You can remove the `.md` at the end, though it shouldn't make a difference).

If you're still having issues, I'll do some more investigation tonight (EST) to see what the issue could be.

**Comment by fkjogu** [comment]
@liamcain, I've just checked that it works, if I remove the file extension `.md`. This might hopefully give you a clue.

Thanks for making this plugin.

**Comment by liamcain** [comment]
@fkjogu Great to hear it's working for you again. I'll investigate why adding the file extension was breaking it.

**Comment by Talor-A** [comment]
still seeing this on `1.2.5`, here are relevant settings:
<img width="1010" alt="Screen Shot 2020-11-17 at 5 28 40 PM" src="https://user-images.githubusercontent.com/11509865/99470933-797b7080-28fa-11eb-95d0-c1415ece28d8.png">
<img width="1010" alt="Screen Shot 2020-11-17 at 5 28 36 PM" src="https://user-images.githubusercontent.com/11509865/99470938-7bddca80-28fa-11eb-9f77-6c6148d7fffa.png">
<img width="1010" alt="Screen Shot 2020-11-17 at 5 29 59 PM" src="https://user-images.githubusercontent.com/11509865/99470970-87c98c80-28fa-11eb-98c9-e61a3f975926.png">

contents of `_templates/daily.md`: 
```
# {{date: MMM D, YYYY}}
```

**Comment by fkjogu** [comment]
>  that it works, if I remove the file extension `.md`

To clarify, both the filename and the entry in the settings have to be updated, obviously.

**Comment by liamcain** [comment]
@fkjogu Aha! That was incredibly helpful, thank you. It sounds like your file was actually named `daily.md.md` on the filesystem. The daily note plugin doesn't seem to have an issue with that so I'll make a fix to be compatible.

@Talor-A Thanks for the detailed report! It sounds like you might have the same root cause as @fkjogu. Could you double check the filename of the template inside your vault to confirm?

I'll try to have a fix for this out sometime today.

**Comment by liamcain** [comment]
Small update: figured out the missing piece on my end, will make a fix for this later today which should hopefully make this a lot more resilient. Thanks for your patience 🙏

**Comment by fkjogu** [comment]
Ah, right. That sounds plausible. Sorry, I wasn't aware that Obsidian swallows the file extensions in its file viewer. Thanks for fixing this!

**Comment by liamcain** [comment]
I just released v1.3.0 which updates the method for finding the daily note template. This should hopefully fix all the issues mentioned in this thread, but I'll reopen if anyone is still having difficulties!
